tests/AddPetForm.test.jsx

```diff
@@ -0,0 +1,107 @@
+    expect(petBirthDateInput).toBeInTheDocument()
+    expect(petBirthDateInput).toHaveAttribute('type', 'date')
+    expect(petBirthDateInput).toBeRequired()
+
+    const lastCheckUpInput = getByLabelText('Last Check-Up Date:')
+    expect(lastCheckUpInput).toBeInTheDocument()
+    expect(lastCheckUpInput).toHaveAttribute('type', 'date')
+
+    const nextCheckUpInput = getByLabelText('Next Check-Up Date:')
+    expect(nextCheckUpInput).toBeInTheDocument()
+    expect(nextCheckUpInput).toHaveAttribute('type', 'date')
+
+    const specialConditionTextarea = getByRole('textbox', { name: 'Special Condition:' })
+    expect(specialConditionTextarea).toBeInTheDocument()
+
+    const formButton = (await findAllByRole('button')).filter(button => button.textContent === 'Add')[0]
+    expect(formButton).toBeInTheDocument()
+    expect(formButton).toHaveAttribute('type', 'submit')
+  })
+
+  it('should send post request to backend on add click with the proper data', async () => {
+    const mockPost = vi.fn().mockResolvedValue({
+      data: {
+        "token": "eyJhbGciOiJIUzI1NiJ9.eyJyb2xlIjoiUk9MRV9VU0VSIiwic3ViIjoidGVzdEBleGFtcGxlLmNvbSIsImlhdCI6MTcyMjkwMDcyOCwiZXhwIjoxNzIyOTAyNTI4fQ.9Fzi6MCTRjyIZ2dSRDVYCWxNZtQbF87THAWJFQ5AT8o",
+        "role": "USER"
+      }
+    })
+    axios.post = mockPost
+
+    const { getByLabelText, findAllByRole } = renderWithRouter(<AddPetForm />)
+
+    const petNameInput = getByLabelText('Name:*')
+    const petBreedInput = getByLabelText('Breed:*')
+    const petSexSelect = getByLabelText('Sex:*')
+    const petBirthDateInput = getByLabelText('Birth Date:*')
+    const lastCheckUpInput = getByLabelText('Last Check-Up Date:')
+    const nextCheckUpInput = getByLabelText('Next Check-Up Date:')
+    const specialConditionTextarea = getByLabelText('Special Condition:')
+    const formButton = (await findAllByRole('button')).filter(button => button.textContent === 'Add')[0]
+
+    fireEvent.change(petNameInput, { target: { value: 'Max' } })
+    fireEvent.change(petBreedInput, { target: { value: 'dog' } })
+    fireEvent.change(petSexSelect, { target: { value: 'Male' } })
+    fireEvent.change(petBirthDateInput, { target: { value: '2024-01-01' } });
+    fireEvent.change(lastCheckUpInput, { target: { value: '2024-06-15' } });
+    fireEvent.change(nextCheckUpInput, { target: { value: '2024-08-15' } });
+    fireEvent.change(specialConditionTextarea, { target: { value: 'No conditions' } });
+
+    fireEvent.click(formButton);
+
+    expect(mockPost).toHaveBeenCalledWith(
+      expect.stringContaining('/user/pet/add'),
+      {
+        "petName": "Max",
+        "petBreed": "dog",
+        "petSex": "Male",
+        "petBirthDate": "2024-01-01",
+        "lastCheckUp": "2024-06-15",
+        "nextCheckUp": "2024-08-15",
+        "specialCondition": "No conditions"
+    },
+      { headers: { 
+        "Content-Type": "application/json",
+        "Authorization": "Bearer null"
```

Comment:
This is looking relatively good, but you might run into an error that this thing is alway `null`. Maybe you wanted to add the token option to it too? If I understand this part should only work if you provide a token.

But if you have a test with `null`, you might have a false positive test.

Reply:
Hi Mark, I tried to mock the token, but somehow it always has Bearer null in the mockpost request. I tried like this:
const mockToken = 'your-mocked-token';
vi.spyOn(localStorage, 'getItem').mockImplementation((key) => {
  if (key === 'token') {
    return mockToken;
  }
  return null;
});

Do you might have any suggestion how I whould do this correctly?